src/node_dns.h

```diff
@@ -0,0 +1,74 @@
+#ifndef SRC_NODE_DNS_H_
+#define SRC_NODE_DNS_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include <unordered_map>
+#include "getdns_ext_libuv.h"
+#include "base_object.h"
+#include "node_mem.h"
+
+namespace node {
+namespace dns {
+
+struct Allocator {
+  void* data;
+  void* (*malloc)(size_t size, void* data);
+  void (*free)(void* ptr, void* data);
+  void* (*calloc)(size_t num, size_t size, void* data);
+  void* (*realloc)(void* ptr, size_t size, void* data);
+};
```

Comment:
Would likely be best to align this with the nghttp2/3 allocators defined in `node_mem.h`